src/lib/bootloader/mbr_update.rb

```diff
@@ -206,25 +206,24 @@ def partition_to_activate(loader_device)
       raise "Invalid loader device #{loader_device}" unless partition
 
       if partition.type == Storage::PartitionType_LOGICAL
-        log.info "Bootloader partition type can be logical"
+        log.info "Bootloader partition cannot be a logical partition, using extended"
         partition = extended_partition(partition)
       end
 
-      log.info "Partition for activating: #{partition}"
+      log.info "Partition for activating: #{partition.inspect}"
       partition
     end
 
     def to_partition(dev_name)
-      device = Y2Storage::BlkDevice.find_by_name(devicegraph, dev_name)
-      if device.is?(:disk)
-        mbr_dev = device
+      device = Y2Storage::Partitionable.find_by_name(devicegraph, dev_name)
+      if device
         # (bnc # 337742) - Unable to boot the openSUSE (32 and 64 bits) after installation
         # if loader_device is disk Choose any partition which is not swap to
         # satisfy such bios (bnc#893449)
-        partition = activatable_partitions(mbr_dev).first
+        partition = activatable_partitions(device).first
         log.info "loader_device is disk device, so use its partition #{partition.inspect}"
       else
-        partition = device
+        partition = nil
```

Comment:
no, it should return Partition class. Just if dev_name is partition name, it should return Partition class and not nil.

calling method have if partition, for devices that is not partition, but data is taken from boot devices and it can look like /dev/sda, /dev/sda1 and in /dev/sda1 case, this method have to return Partition class and not nil.